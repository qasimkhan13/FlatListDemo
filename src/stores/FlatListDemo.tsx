import React, {Component} from 'react';
import { StyleSheet, Text, View} from 'react-native';
import { FlatList } from 'react-native';


export default class FlatListDemo extends Component { 
  render() {
    return (
      <View style={styles.container}>
                <FlatList
                    data={[{ key: 'a' }, { key: 'b' }]}
                    renderItem={(item:any) => <Text>{item.key}</Text>}
                />   
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
