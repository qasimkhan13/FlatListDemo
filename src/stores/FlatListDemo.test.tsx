import React, {Component} from 'react';
import { mount } from "enzyme";
import FlatListDemo from './FlatListDemo'
describe("Mount ", () => {
  
  it("mount flatlist", () => {

    const wrapper:any = mount(
      <FlatListDemo/> 
      )
  console.log('wrapper', wrapper.props())
      expect(wrapper).toMatchSnapshot();
      expect(wrapper).toBeDefined()
  });


});
