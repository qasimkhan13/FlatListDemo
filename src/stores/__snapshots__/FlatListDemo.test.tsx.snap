// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mount  mount flatlist 1`] = `
<FlatListDemo>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <FlatList
        data={
          Array [
            Object {
              "key": "a",
            },
            Object {
              "key": "b",
            },
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      >
        <VirtualizedList
          data={
            Array [
              Object {
                "key": "a",
              },
              Object {
                "key": "b",
              },
            ]
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <ScrollView
            data={
              Array [
                Object {
                  "key": "a",
                },
                Object {
                  "key": "b",
                },
              ]
            }
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <_class
              data={
                Array [
                  Object {
                    "key": "a",
                  },
                  Object {
                    "key": "b",
                  },
                ]
              }
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <RCTScrollView
                data={
                  Array [
                    Object {
                      "key": "a",
                    },
                    Object {
                      "key": "b",
                    },
                  ]
                }
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <Component>
                  <View>
                    <CellRenderer
                      cellKey="a"
                      fillRateHelper={
                        FillRateHelper {
                          "_anyBlankStartTime": null,
                          "_enabled": false,
                          "_getFrameMetrics": [Function],
                          "_info": Info {
                            "any_blank_count": 0,
                            "any_blank_ms": 0,
                            "any_blank_speed_sum": 0,
                            "mostly_blank_count": 0,
                            "mostly_blank_ms": 0,
                            "pixels_blank": 0,
                            "pixels_sampled": 0,
                            "pixels_scrolled": 0,
                            "sample_count": 0,
                            "total_time_spent": 0,
                          },
                          "_mostlyBlankStartTime": null,
                          "_samplesStartTime": null,
                        }
                      }
                      horizontal={false}
                      index={0}
                      inversionStyle={null}
                      item={
                        Object {
                          "key": "a",
                        }
                      }
                      key="a"
                      onLayout={[Function]}
                      onUnmount={[Function]}
                      onUpdateSeparators={[Function]}
                      parentProps={
                        Object {
                          "data": Array [
                            Object {
                              "key": "a",
                            },
                            Object {
                              "key": "b",
                            },
                          ],
                          "disableVirtualization": false,
                          "getItem": [Function],
                          "getItemCount": [Function],
                          "horizontal": false,
                          "initialNumToRender": 10,
                          "keyExtractor": [Function],
                          "maxToRenderPerBatch": 10,
                          "numColumns": 1,
                          "onEndReachedThreshold": 2,
                          "renderItem": [Function],
                          "scrollEventThrottle": 50,
                          "updateCellsBatchingPeriod": 50,
                          "viewabilityConfigCallbackPairs": Array [],
                          "windowSize": 21,
                        }
                      }
                    >
                      <Component
                        onLayout={[Function]}
                        style={null}
                      >
                        <View
                          onLayout={[Function]}
                          style={null}
                        >
                          <Component>
                            <Text />
                          </Component>
                        </View>
                      </Component>
                    </CellRenderer>
                    <CellRenderer
                      cellKey="b"
                      fillRateHelper={
                        FillRateHelper {
                          "_anyBlankStartTime": null,
                          "_enabled": false,
                          "_getFrameMetrics": [Function],
                          "_info": Info {
                            "any_blank_count": 0,
                            "any_blank_ms": 0,
                            "any_blank_speed_sum": 0,
                            "mostly_blank_count": 0,
                            "mostly_blank_ms": 0,
                            "pixels_blank": 0,
                            "pixels_sampled": 0,
                            "pixels_scrolled": 0,
                            "sample_count": 0,
                            "total_time_spent": 0,
                          },
                          "_mostlyBlankStartTime": null,
                          "_samplesStartTime": null,
                        }
                      }
                      horizontal={false}
                      index={1}
                      inversionStyle={null}
                      item={
                        Object {
                          "key": "b",
                        }
                      }
                      key="b"
                      onLayout={[Function]}
                      onUnmount={[Function]}
                      onUpdateSeparators={[Function]}
                      parentProps={
                        Object {
                          "data": Array [
                            Object {
                              "key": "a",
                            },
                            Object {
                              "key": "b",
                            },
                          ],
                          "disableVirtualization": false,
                          "getItem": [Function],
                          "getItemCount": [Function],
                          "horizontal": false,
                          "initialNumToRender": 10,
                          "keyExtractor": [Function],
                          "maxToRenderPerBatch": 10,
                          "numColumns": 1,
                          "onEndReachedThreshold": 2,
                          "renderItem": [Function],
                          "scrollEventThrottle": 50,
                          "updateCellsBatchingPeriod": 50,
                          "viewabilityConfigCallbackPairs": Array [],
                          "windowSize": 21,
                        }
                      }
                      prevCellKey="a"
                    >
                      <Component
                        onLayout={[Function]}
                        style={null}
                      >
                        <View
                          onLayout={[Function]}
                          style={null}
                        >
                          <Component>
                            <Text />
                          </Component>
                        </View>
                      </Component>
                    </CellRenderer>
                  </View>
                </Component>
              </RCTScrollView>
            </_class>
          </ScrollView>
        </VirtualizedList>
      </FlatList>
    </View>
  </Component>
</FlatListDemo>
`;
